File: src/main/java/com/robohorse/robopojogenerator/view/GeneratorVew.java
package com.robohorse.robopojogenerator.view;

import org.fife.ui.rsyntaxtextarea.RSyntaxTextArea;
import org.fife.ui.rsyntaxtextarea.SyntaxConstants;
import org.fife.ui.rsyntaxtextarea.Theme;

import javax.swing.*;
import java.io.IOException;

/**
 * Created by vadim on 24.09.16.
 */
public class GeneratorVew {
    private JPanel rootView;
    private JButton generateButton;
    private RSyntaxTextArea textArea;
    private JCheckBox rewriteExistingClassesCheckBox;
    private JTextField className;
    private JScrollPane jsonAreaScrollView;
    private JRadioButton JSONRadioButton;
    private JRadioButton JSONSchemaRadioButton;
    private JRadioButton javaRadioButton;
    private JRadioButton kotlinRadioButton;
    private JPanel sourcePanel;
    private JPanel languagePanel;
    private JPanel frameworkPanel;
    private JList<String> frameworkList;
    private JPanel actionPanel;
    private JPanel propertiesPanel;
    private JPanel commonInfoPanel;
    private JPanel controlsPanel;
    private JScrollPane scrollPropertiesPanel;
    private JCheckBox useTabsIndentation;
    private ButtonGroup sourceGroup;
    private ButtonGroup languageGroup;

    public JCheckBox getUseTabsIndentation() {
        return useTabsIndentation;
    }

    public JRadioButton getJavaRadioButton() {
        return javaRadioButton;
    }

    public JPanel getSourcePanel() {
        return sourcePanel;
    }

    public JRadioButton getJSONRadioButton() {
        return JSONRadioButton;
    }

    public JRadioButton getJSONSchemaRadioButton() {
        return JSONSchemaRadioButton;
    }

    public JPanel getPropertiesPanel() {
        return propertiesPanel;
    }

    public JRadioButton getKotlinRadioButton() {
        return kotlinRadioButton;
    }

    public JPanel getRootView() {
        return rootView;
    }

    public JList<String> getFrameworkList() {
        return frameworkList;
    }

    public JTextField getClassNameTextField() {
        return className;
    }

    public JButton getGenerateButton() {
        return generateButton;
    }

    public JTextArea getTextArea() {
        return textArea;
    }

    public JTextField getClassName() {
        return className;
    }

    public JCheckBox getRewriteExistingClassesCheckBox() {
        return rewriteExistingClassesCheckBox;
    }

    private void createUIComponents() {
        textArea = new RSyntaxTextArea();
        textArea.setSyntaxEditingStyle(SyntaxConstants.SYNTAX_STYLE_JSON);
        textArea.setCodeFoldingEnabled(true);
        jsonAreaScrollView = new JScrollPane(textArea);
        try {
            final Theme theme = Theme.load(getClass().getResourceAsStream(
                    "/org/fife/ui/rsyntaxtextarea/themes/monokai.xml"));
            theme.apply(textArea);
        } catch (IOException ioe) {
            ioe.printStackTrace();
        }
    }
}

File: app/src/main/kotlin/com/robohorse/robopojogenerator/form/GeneratorVew.kt
package com.robohorse.robopojogenerator.form

import org.fife.ui.rsyntaxtextarea.RSyntaxTextArea
import org.fife.ui.rsyntaxtextarea.SyntaxConstants
import org.fife.ui.rsyntaxtextarea.Theme
import java.io.IOException
import javax.swing.ButtonGroup
import javax.swing.JButton
import javax.swing.JCheckBox
import javax.swing.JList
import javax.swing.JPanel
import javax.swing.JRadioButton
import javax.swing.JScrollPane
import javax.swing.JTextField

/**
 * Created by vadim on 24.09.16.
 */
class GeneratorVew {
    lateinit var rootView: JPanel
    lateinit var generateButton: JButton
    lateinit var textArea: RSyntaxTextArea
    lateinit var rewriteExistingClassesCheckBox: JCheckBox
    lateinit var className: JTextField
    lateinit var jsonAreaScrollView: JScrollPane
    lateinit var JSONRadioButton: JRadioButton
    lateinit var JSONSchemaRadioButton: JRadioButton
    lateinit var javaRadioButton: JRadioButton
    lateinit var kotlinRadioButton: JRadioButton
    lateinit var sourcePanel: JPanel
    lateinit var languagePanel: JPanel
    lateinit var frameworkPanel: JPanel
    lateinit var frameworkList: JList<String>
    lateinit var actionPanel: JPanel
    lateinit var propertiesPanel: JPanel
    lateinit var commonInfoPanel: JPanel
    lateinit var controlsPanel: JPanel
    lateinit var scrollPropertiesPanel: JScrollPane
    lateinit var useTabsIndentation: JCheckBox
    lateinit var sourceGroup: ButtonGroup
    lateinit var languageGroup: ButtonGroup

    private fun createUIComponents() {
        textArea = RSyntaxTextArea()
        textArea.syntaxEditingStyle = SyntaxConstants.SYNTAX_STYLE_JSON
        textArea.isCodeFoldingEnabled = true
        jsonAreaScrollView = JScrollPane(textArea)
        try {
            val theme = Theme.load(
                javaClass.getResourceAsStream(
                    "/org/fife/ui/rsyntaxtextarea/themes/monokai.xml"
                )
            )
            theme.apply(textArea)
        } catch (ioe: IOException) {
            ioe.printStackTrace()
        }
    }
}
